Src/orbtraceIf.c

```diff
@@ -472,8 +473,54 @@ bool OrbtraceGetIfandEP( struct OrbtraceIf *o )
             return false;
 
         case DEVICE_BMP: // -----------------------------------------------------------------------------
-            o->iface = BMP_IFACE;
-            o->ep = BMP_EP;
+            genericsReport( V_DEBUG, "Searching for BMP trace interface" EOL );
+
+            if ( ( err = libusb_get_active_config_descriptor( o->dev, &config ) ) < 0 )
+            {
+                genericsReport( V_WARN, "Failed to get config descriptor (%d)" EOL, err );
+                return false;
+            }
+
+            /* Loop through the interfaces looking for ours */
+            for ( int if_num = 0; if_num < config->bNumInterfaces && !interface_found; if_num++ )
```

Comment:
Do note that `config->bNumInterfaces` is a uint8_t, and so this does signed-unsigned conversion which is best avoided.

Reply:
Aye, they're review items for things we'd do differently but we are not the Orbuculum maintainers (we are only the lead maintainer for BMD, so our review was purely about making things robust against BMD) which is why we approved with comments - it's up to Mubes and zyp if they want these items changed/addressed but we wouldn't have felt right not pointing them out to be able to have a conversation about.

Reply:
I guess for this PR it should be ok, and there should be another PR. I will make new one, once this is merged and time permits. Executing cppcheck on the orbuculum codebase revelals more problems.